packages/agreement-process/src/services/agreementService.ts

```diff
@@ -734,3 +757,75 @@ export async function cloneAgreementLogic({
     events: [newAgreement, ...docEvents],
   };
 }
+
+export async function rejectAgreementLogic({
+  agreementId,
+  rejectionReason,
+  authData,
+  agreementQuery,
+  tenantQuery,
+  eserviceQuery,
+}: {
+  agreementId: string;
+  rejectionReason: string;
+  authData: AuthData;
+  agreementQuery: AgreementQuery;
+  tenantQuery: TenantQuery;
+  eserviceQuery: EserviceQuery;
+}): Promise<CreateEvent<AgreementEvent>> {
+  const agreementToBeRejected = await agreementQuery.getAgreementById(
+    agreementId
+  );
+  assertAgreementExist(agreementId, agreementToBeRejected);
+
+  assertRequesterIsProducer(agreementToBeRejected.data, authData);
+
+  assertExpectedState(agreementId, agreementToBeRejected.data.state, [
+    agreementState.pending,
+  ]);
+
+  const eservice = await eserviceQuery.getEServiceById(
+    agreementToBeRejected.data.eserviceId
+  );
+  assertEServiceExist(agreementToBeRejected.data.eserviceId, eservice);
+
+  const consumer = await tenantQuery.getTenantById(
+    agreementToBeRejected.data.consumerId
+  );
+  assertTenantExist(agreementToBeRejected.data.consumerId, consumer);
+
+  const descriptor = eservice.data.descriptors.find(
+    (d) => d.id === agreementToBeRejected.data.descriptorId
+  );
+  assertDescriptorExist(
+    eservice.data.id,
+    agreementToBeRejected.data.descriptorId,
+    descriptor
+  );
+
+  const stamp = createStamp(authData);
+  const rejected: Agreement = {
+    ...agreementToBeRejected.data,
+    state: agreementState.rejected,
+    certifiedAttributes: matchingCertifiedAttributes(descriptor, consumer.data),
+    declaredAttributes: matchingDeclaredAttributes(descriptor, consumer.data),
+    verifiedAttributes: matchingVerifiedAttributes(
+      eservice.data,
+      descriptor,
+      consumer.data
+    ),
+    rejectionReason,
+    suspendedByConsumer: undefined,
+    suspendedByProducer: undefined,
+    suspendedByPlatform: undefined,
+    stamps: {
+      ...agreementToBeRejected.data.stamps,
+      rejection: stamp,
+    },
```

Comment:
Missing `consumerNotes: agreementToBeRejected.consumerNotes` ?

<img width="772" alt="image" src="https://github.com/pagopa/interop-be-monorepo/assets/6418684/11270fcd-0dbe-4704-ac9a-91f53f152a38">